lti_consumer/lti_1p3/extensions/rest_framework/serializers.py

```diff
@@ -227,3 +230,90 @@ class Meta:
             'resultMaximum',
             'comment',
         )
+
+
+# pylint: disable=abstract-method
+class LtiContextSerializer(serializers.Serializer):
+    """
+    Serializer for a LTI Context
+    """
+    id = UsageKeyField()
+
+
+class LtiNrpsContextMemberBasicSerializer(serializers.Serializer):
+    """
+    Non PII fields serializer for Context Member.
+    """
+    status = serializers.SerializerMethodField()
+    user_id = serializers.SerializerMethodField()
+    roles = serializers.SerializerMethodField()
+
+    def get_user_id(self, user):
+        """
+        Get external user id for the user.
+        """
+        externalid, _ = compat.get_or_create_externalid(user)
+        return serializers.UUIDField().to_representation(externalid.external_user_id)
+
+    def get_status(self, user):
+        """
+        Get user status string value.
+        """
+        return 'Active' if user.is_active else 'Inactive'
+
+    def get_roles(self, user):
+        """
+        Prepare and return Context Roles for user.
+        """
+        roles = []
+        if user.courseenrollment_set.count() > 0:
+            roles += LTI_1P3_CONTEXT_ROLE_MAP['student']
+
+        for accessrole in user.courseaccessrole_set.all():
+            if LTI_1P3_CONTEXT_ROLE_MAP.get(accessrole.role):
+                roles += LTI_1P3_CONTEXT_ROLE_MAP[accessrole.role]
+
+        return set(roles)
+
+
+class LtiNrpsContextMemberPIISerializer(LtiNrpsContextMemberBasicSerializer):
+    """
+    Personally identifiable information serializer for Context Member.
+    """
+    name = serializers.CharField(source='profile.name', required=False)
+    picture = serializers.SerializerMethodField()
+    email = serializers.EmailField(required=False)
+
+    def get_picture(self, user):
+        """
+        Get absolute url of user's profile picture.
+        """
+        request = self.context['request']
+        picture_url = compat.get_user_profile_image(user)['medium']
```

Comment:
Again, this can cause a query "explosion", and needs to be prefetched in the queryset.
If there's no easy way, you can leave this parameter out for now.

Reply:
@giovannicimolin, I don't think it will cause a query "explosion" as the ``openedx.core.djangoapps.user_api.accounts.image_helpers.get_profile_image_urls_for_user`` doesn't do any query. It will use the image storage to retrieve image URLs and generate names using a hash function.

Reply:
@shimulch This method actually uses the `user.profile` related model, which will cause trouble if you're not prefetching `profile` in the queryset. Add a test that counts queries on this and you'll see what I mean. If there's more users, more queries will be performed.